src/NServiceBus.Core/Sagas/SagaMessage.cs

```diff
@@ -10,16 +12,22 @@ public class SagaMessage
         /// </summary>
         /// <param name="messageType">Type of the message.</param>
         /// <param name="isAllowedToStart"><code>true</code> if the message can start the saga, <code>false</code> otherwise.</param>
-        public SagaMessage(string messageType, bool isAllowedToStart)
+        public SagaMessage(Type messageType, bool isAllowedToStart)
         {
             MessageType = messageType;
+            MessageTypeName = messageType.FullName;
             IsAllowedToStartSaga = isAllowedToStart;
         }
 
         /// <summary>
         /// The type of the message.
         /// </summary>
-        public string MessageType { get; private set; }
+        public Type MessageType { get; private set; }
```

Comment:
I'm on the fence on this one. I think we should not touch it before the
next round of saga improvements. The value it provides is not enough to
justify the change. Thoughts?

On Saturday, May 21, 2016, David Boike <notifications@github.com> wrote:

> In src/NServiceBus.Core/Sagas/SagaMessage.cs
> <https://github.com/Particular/NServiceBus/pull/3756#discussion_r64133390>
> :
>
> >              IsAllowedToStartSaga = isAllowedToStart;
> >          }
> >
> >          /// <summary>
> >          /// The type of the message.
> >          /// </summary>
> > -        public string MessageType { get; private set; }
> > +        public Type MessageType { get; private set; }
>
> Having the type allows me to compare on type, obviously, but then also
> gives me the ability to use the simple type name in exception messages to
> make them really useful and prescriptive.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3756/files/a7617ae55a797e75be74b7b9b8d8d329900c7d6e#r64133390>
>